### ISSUE TITLE
USWDS - Bug: Responsive stacked table cannot be parsed by VoiceOver on Safari

### ISSUE THREAD

**Issue by devowhippit** [uswds/uswds#5434]
### Describe the bug

I was looking into using the [responsive stacked table](https://designsystem.digital.gov/components/table/#responsive-stacked-table-2) this week, and I found that I couldn't use VoiceOver on Safari to read the table contents at all. I was wondering if I was using voice-over controls wrong, but after looking into it a bit more, it seems like a documented issue (I was not aware of before) that modifying the display properties of tables makes them inoperable with this SR+Browser combination.

Using Safari v16.5.2 on MacOS.

[Bug and screen-reader support reference and documentation](https://adrianroselli.com/2022/07/its-mid-2022-and-browsers-mostly-safari-still-break-accessibility-via-display-properties.html).

### Steps to reproduce the bug

1. Navigate to a [responsive stacked table example](https://designsystem.digital.gov/components/table/#responsive-stacked-table-2) using a MacOS + Safari
2. Use keyboard commands VO (ctrl + opt) + Left Arrow to navigate to the table example. VoiceOver will announce the table caption and display a visible focus indicator on the table.
3. VoiceOver will instruct to use VO (ctrl + opt) + Shift + Down Arrow to enter into the table. 
4. When entering into the table, however, VoiceOver will announce the table as empty, and the cells can't be navigated. 

### Expected Behavior

At step 4, the user should be able to navigate the table cells using VO (ctrl + opt) + arrow keys.

### Screenshots

![Screenshot 2023-08-10 at 2 19 10 PM](https://github.com/uswds/uswds/assets/3826974/7a898a2c-38b1-41ce-afac-7fca9f51683d)
![Screenshot 2023-08-10 at 2 19 41 PM](https://github.com/uswds/uswds/assets/3826974/9444cc11-29b5-4c26-882a-72988f124825)

### System setup

- Using Safari v16.5.2 on MacOS
- USWDS v3.5
- USWDS public documentation site

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [X] I checked the [current issues](https://github.com/uswds/uswds/issues?q=is%3Aissue+is%3Aopen+label%3A%22Type%3A+Bug%22) for duplicate bug reports.

**Comment by mejiaj** [comment]
@mahoneycm has confirmed that this is an issue - additional discussion in [slack](https://gsa-tts.slack.com/archives/C3F14AHSQ/p1691684085796299).

**Comment by mejiaj** [comment]
Leaving in 3.7.0 to spike and research potential options.

>**Note**
>If LOE is too high for sprint we can move it to next release.
